src/inmanta/protocol/rest/__init__.py

```diff
@@ -241,21 +241,35 @@ async def _get_dict_value_from_message(
         # Check if the values should be converted to lists
         type_args = typing_inspect.get_args(self._argspec.annotations.get(arg), evaluate=True)
         if issubclass(type_args[1], list):
-            value = {key: val.split(",") if not isinstance(val, list) else val for key, val in value.items()}
+            value = {
+                key: self._get_list_value_from_url(val) if not isinstance(val, list) else val for key, val in value.items()
+            }
         return value
 
     def _is_dict_or_optional_dict(self, arg_type: Type) -> bool:
         if typing_inspect.is_optional_type(arg_type):
             arg_type = typing_inspect.get_args(arg_type, evaluate=True)[0]
         return issubclass(arg_type, dict)
 
+    def _get_list_value_from_url(self, value: str) -> List[str]:
+        # Split by the delimiter
+        split_parts = value.split(",")
+        empty_strings = [part for part in split_parts if len(part) == 0]
+        if len(empty_strings):
+            raise exceptions.BadRequest(
```

Comment:
Doesn't this mean we are not correctly url-encoding all the values? 

I would like this encoding issue to be fixed, such that we can transport all values correctly. 

What if the value contains `%20` does it come out as a space of as `%20`  ?